api/METRICS.md

```diff
@@ -0,0 +1,114 @@
+# API proposal for metrics
+
+## Example
+
+```python
+# For most sklearn metrics, we will have their group version that returns
+# the summary of its performance across groups as well as the overall
+# performance, represented as a Bunch object with fields
+#   * overall: overall metric value
+#   * by_group: a dictionary that maps sensitive feature values to metric values
+
+summary = accuracy_score_group_summary(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+
+# Exporting into pd.Series or pd.DataFrame in not too complicated
+
+series = pd.Series({**summary.by_group, 'overall': summary.overall})
+df = pd.DataFrame({"model accuracy": {**summary.by_group, 'overall': summary.overall}})
+
+# Several types of scalar metrics for group fairness can be obtained from the group summary via transformation functions
+
+acc_difference = difference_from_summary(summary)
+acc_ratio = ratio_from_summary(summary)
+acc_group_min = group_min_from_summary(summary)
+
+# Most common disparity metrics should be predefined
+
+demo_parity_difference = demographic_parity_difference(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+demo_parity_ratio = demographic_parity_ratio(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+eq_odds_difference = equalized_odds_difference(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+
+# For predefined disparities based on sklearn metrics, we adopt a consistent naming conventions
+
+acc_difference = accuracy_score_difference(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+acc_group_min = accuracy_score_group_min(y_true, y_pred, sensitive_features=sf, **other_kwargs)
+```
+
+## Proposal
+
+*Function signatures*
+
+```python
+group_summary(metric, y_true, y_pred, *, sensitive_features, **other_kwargs)
+# return the group summary for the provided `metric`, where `metric` has the signature
+# metric(y_true, y_pred, **other_kwargs)
+
+make_metric_group_summary(metric)

```

Comment:
OK, although for strict comparison to `make_scorer` it should be `make_metric_group_summariser()`

Reply:
You mean `make_metric_group_summarizer()` ;-)

Reply:
my rationale for `make_metric_group_summary` is that the result is the function to which we will refer to as `<metric>_group_summary` elsewhere in the API; that in turn makes sense since other derived metrics are called `<metric>_difference`, `<metric>_ratio` etc.

Reply:
OK